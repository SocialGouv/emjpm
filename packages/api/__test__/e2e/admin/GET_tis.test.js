//

const request = require("supertest");

const { knex } = global;
jest.setMock("@emjpm/api/db/knex", knex);

const server = require("@emjpm/api/app");

const { getTokenByUserType } = require("../utils");

//

beforeAll(async () => {
  await knex.migrate.latest();
  await knex.seed.run();
});

// strip the created_at date in the response

// eslint-disable-next-line no-unused-vars
const simpler = ({ created_at, ...props }) => props;

test("should GET tis", async () => {
  const token = await getTokenByUserType("admin");
  const response = await request(server)
    .get("/api/v1/admin/tis")
    .set("Authorization", "Bearer " + token);

  expect(response.body.map(simpler)).toMatchSnapshot();

  expect(response.status).toBe(200);
});
