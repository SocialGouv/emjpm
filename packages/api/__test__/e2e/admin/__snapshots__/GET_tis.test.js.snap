// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should GET tis 1`] = `
Array [
  Object {
    "active": true,
    "cabinet": "6D",
    "email": "test2@ti.com",
    "id": 5,
    "last_login": null,
    "nom": null,
    "prenom": "Pierre",
    "type": "ti",
  },
  Object {
    "active": false,
    "cabinet": null,
    "email": null,
    "id": 7,
    "last_login": null,
    "nom": null,
    "prenom": "Jack",
    "type": "ti",
  },
  Object {
    "active": false,
    "cabinet": "6D",
    "email": "test@ti.com",
    "id": 32,
    "last_login": null,
    "nom": null,
    "prenom": "Jean",
    "type": "ti",
  },
  Object {
    "active": true,
    "cabinet": "6D",
    "email": "test3@ti.com",
    "id": 33,
    "last_login": null,
    "nom": null,
    "prenom": "Momo",
    "type": "ti",
  },
]
`;
